climada/hazard/base.py

```diff
@@ -277,9 +277,16 @@ def from_raster(cls, files_intensity, files_fraction=None, attrs=None,
         if files_fraction is not None and len(files_intensity) != len(files_fraction):
             raise ValueError('Number of intensity files differs from fraction files: %s != %s'
                              % (len(files_intensity), len(files_fraction)))
-        if haz_type is None:
-            haz_type = cls().tag.haz_type
-        haz = cls(haz_type, pool)
+
+        if haz_type and cls is Hazard:
+            haz = cls(haz_type=haz_type, pool=pool)
+        else:
+            haz = cls(pool=pool)
```

Comment:
I am not sure this is optimal. This will break as soon as someone makes a hazard with `haz_type` as argument right? Maybe this is very unlikely though...

Reply:
As I said in CLIMADA-project/climada_petals#15, there are infinitely many ways how any of this can fail. This PR is supposed to be a fix for the current state of the code *as is*. In a future version, we need to require new child classes to implement unit tests also for the inherited (and not only for the added) methods. In that case, this kind of problem can be discussed once it is actually introduced.

Reply:
I agree with @tovogt and I have some doubts about how well that will go if someone introduces haz_type as truly variable in a Hazard class. However - alternatively one can just _try_ to use the haz_type as an argument in the constructor. (See suggestion.)
Any other thinkable deviation of the constructor pattern for subclasses of  Hazard, `def __init__(self, pool)`, will still break.